scripts/fuzz.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+set -e
+
+# Simple wrapper around a libfuzzer test run, as much for
+# documentation as direct use.  The idea here is really simple: build
+# for the Zephyr "native_posix" board (which is a just a x86
+# executable for the build host, not an emulated device) and run the
+# resulting zephyr.exe file.  This specifies a "fuzz_corpus" directory
+# to save the seeds that produce useful coverage output for use in
+# repeated runs (these are not particularly large, we might consider
+# curating and commiting such a seed directory to the tree).
+#
+# The tool will run until it finds a failure condition.  You will see
+# MANY errors on stdout from all the randomized input.  Don't try to
+# capture this, either let it output to a terminal or arrange to keep
+# only the last XXX lines after the tool exits.
+#
+# The only prerequisite to install is a clang compiler on the host.
+# Versions 12+ have all been observed to work.
+#
+# You will need the kconfigs specified below for correct operation,
+# but can add more at the end of this script's command line to
+# duplicate configurations as needed.
+
+export ZEPHYR_BASE=$SOF_TOP/../zephyr
+export ZEPHYR_TOOLCHAIN_VARIANT=llvm
+
+west build -p -b native_posix $SOF_TOP/app/ -- \
+  -DCONFIG_ASSERT=y \
+  -DCONFIG_SYS_HEAP_BIG_ONLY=y \
+  -DCONFIG_ZEPHYR_NATIVE_DRIVERS=y \
+  -DCONFIG_ARCH_POSIX_LIBFUZZER=y \
+  -DCONFIG_ARCH_POSIX_FUZZ_TICKS=100 \
+  -DCONFIG_ASAN=y "$@"
```

Comment:
Would it make sense to have all these in some kind of `defconfig` file? Overkill / too cumbersome?

Reply:
I defer to you guys on that?  There's a thousand ways to do kconfig and this is the one that steps on the fewest toes.  Another option would be to create a new kconfig like CONFIG_SOF_FUZZER that then selects this stuff.

(Also I'm noticing that heap one, which doesn't belong.  I forget the details now, but the SOF "posix" platform creates a heap of the same size as the hardware platforms, but for some reason ends up building a zephyr without support for the "big" heap code and so gets runtime errors.  It's just a kconfig mismatch somewhere I worked around and then forgot about.)

Reply:
@kv2019i any comment on this re our Zephyr configs ?

Reply:
I think it's fine here. An overlay file like sof/app/perf_overlay.conf (passed via -DOVERLAY_CONFIG= to west) would be perhaps be a good match, but as long as the assumption is the users all use this helper script anyways, I think the definitions are fine here.

Reply:
> An overlay file like sof/app/perf_overlay.conf (passed via -DOVERLAY_CONFIG= to west) would be perhaps be a good match

@andyross can you add this as a one-line comment? Thx!